src/main/java/net/mcreator/element/converter/v2024_2/PotionEffectCategoryConverter.java

```diff
@@ -29,15 +29,15 @@
 import org.apache.logging.log4j.Logger;
 
 public class PotionEffectCategoryConverter implements IConverter {
-
 	private static final Logger LOG = LogManager.getLogger(PotionEffectCategoryConverter.class);
 
 	@Override
 	public GeneratableElement convert(Workspace workspace, GeneratableElement input, JsonElement jsonElementInput) {
+		PotionEffect potionEffect = (PotionEffect) input;
 		try {
 			JsonObject definition = jsonElementInput.getAsJsonObject().getAsJsonObject("definition");
 
-			if(input instanceof PotionEffect potion) {
+			if(potionEffect instanceof PotionEffect potion) {

```

Comment:
There should be no need to cast an object of one class to the same class.

Reply:
IDEA even shows this as a warning so not sure how this went through

Reply:
Fixed